pkg/controller/provider/container/ovirt/client.go

```diff
@@ -73,15 +93,53 @@ func (r *Client) connect() (err error) {
 			TLSClientConfig:       TLSClientConfig,
 		},
 	}
+
+	url, err := liburl.Parse(r.url)
+	if err != nil {
+		return
+	}
+
+	url.Path = "/ovirt-engine/sso/oauth/token"
+	values := url.Query()
+	values.Add("grant_type", "password")
+	values.Add("username", string(r.secret.Data["user"]))
+	values.Add("password", string(r.secret.Data["password"]))
+	values.Add("scope", "ovirt-app-api")
+
 	client.Header = http.Header{
 		"Accept": []string{"application/json"},
-		"Authorization": []string{
-			"Basic",
-			r.auth()},
-		"Version": []string{"4"},
+	}
+
+	response := &ovirtTokenResponse{}
+	url.RawQuery = values.Encode()
+	status, err := client.Get(url.String(), response)
+	if err != nil {
+		return
+	}
+
+	// Providing bad credentials when requesting the token results
+	// in 400, and not 401. So checking for != 200 instead
+	if status != http.StatusOK {
+		err = liberr.New("Request for token failed", "status", status)
+		return
+	}
+
+	// Set the access token we received
+	client.Header = http.Header{
+		"Accept":        []string{"application/json"},
+		"Authorization": []string{"Bearer " + response.AccessToken},
+		"Version":       []string{"4"},
 	}
 
 	r.client = client
+	r.created = time.Now()
+
+	expiration, err := strconv.ParseInt(response.Expiration, 10, 64)
+	if err != nil {
+		return
```

Comment:
looking at the error Martin mentioned below, if we return that error back I'm afraid it won't be so clear to users - how about returning an error with better description, i.e., saying that we failed to parse the expiration part